tools/find-missing-package-kdl.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/sh
+# List component directories that do not have a package.kdl file.
+# A "component directory" is defined here as any directory under
+#   components/ that contains a Makefile (case-sensitive name Makefile).
+# Output:
+#   - Prints each component directory (relative path) missing package.kdl
+#   - At the end prints a summary count
+# Exit code:
+#   - Always 0 (for easy piping/consumption)
+
+set -eu
+
+# Move to repo root if invoked from elsewhere
+ROOT_DIR=$(cd "$(dirname "$0")/.." && pwd)
+cd "$ROOT_DIR"
+
+# Collect component root directories (heuristic):
```

Comment:
For list of component directories you can use `components.mk`.

Reply:
I wanted something that does not rely on the setup target to find components with issues.